api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -543,6 +563,122 @@ public int hashCode() {
     }
   }
 
+  public static class GeometryType extends PrimitiveType {
+
+    private final String crs;
+
+    private GeometryType(String crs) {
+      Preconditions.checkNotNull(crs, "CRS cannot be null");
```

Comment:
Style: In Iceberg, we try to have direct and clear error messages by cutting out sentence structure. Here, we would normally use `"Invalid CRS: null"`.

Reply:
We allow `null` but checks for empty string and leading/trailing spaces. The error messages are:

* `Invalid CRS: (empty string)` for empty string
* `CRS must not have leading or trailing spaces: '%s'` for CRS with leading or trailing spaces.